### PR TITLE
Replace error chain

### LINKED ISSUE #24: Replace error-chain
We should replace `error-chain` with custom error type allowing us to provide more precise error messages like which part of the expression is wrong, etc.

### PR DESCRIPTION
**Pull request by zrzka**
Fixes #24 

PR contains:

* removed explicit notifications from `.travis.yml`
* added @cyplo as an author
* replaced error-chain with custom error type [1]
* integrations tests imports organized

[1] This is ongoing work, because I'd like to add more info to error messages (like the whole expression, etc.), but it's kinda okay for now. It's nothing that should block us - internal stuff, no public API breakage.

Why custom error and not failure for example? error-chain was deprecated and failure recommended. The problem here is that even the failure is not magic & final solution, there's still some ongoing work, ... Also I think that we should use custom error types for libs, which do comply with `std::error::Error`. The reason I was using error-chain was simple - speed & experimentation. Now, when things are settling down, there's no reason to use it.

### CODE REVIEW

File: src/builtin/filter/datetime.rs [hunk 1]
```diff
@@ -4,17 +4,29 @@ use chrono::{DateTime, NaiveDateTime, Utc};
...
+            .context("input", input.to_string())
+    })?;
 
     let format = match args.get("format") {
-        Some(x) => x
-            .as_str()
-            .ok_or_else(|| format!("`{}` format must be a string", filter))?,
+        Some(x) => x.as_str().ok_or_else(|| {
+            Error::with_message("invalid argument type")
+                .context("filter", filter)
```
Comment (cyplo) [inline src/builtin/filter/datetime.rs]: I like this, ability to add context fluently

File: src/error.rs [hunk 2]
```diff
@@ -1,16 +1,158 @@
...
+        K: Into<Key>,
+        V: Into<Value>,
+    {
+        self.map_err(|e| e.context(k, v))
+    }
+}
+
+/// Error type
+pub struct Error {
+    inner: Box<Inner>,
```
Comment (cyplo) [inline src/error.rs]: Is box needed because we expect the error to be rather big ?
If so - the comment could help readers here, to list examples in which box could be beneficial ?
If not - we could potentially get rid of `Inner` and just have `Error` with all the fields ? What do you think ? (I removed the Box to try and everything seems happy)

File: src/error.rs [hunk 3]
```diff
@@ -1,16 +1,158 @@
...
+        self.inner.push(k, v);
+        self
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        writeln!(f, "temen: {}", self.inner.message)?;
+        if !self.inner.context.is_empty() {
+            writeln!(f, " └ context:")?;
```
Comment (cyplo) [inline src/error.rs]: this is pretty cool :)

File: src/builtin/filter/datetime.rs (hunk 1 above)
Comment (cyplo) [reply src/builtin/filter/datetime.rs]: On some other note - I think (later?) it could help to showcase how cool this is, with some doc-tests maybe ? It may not be that easy to discover how powerful the usage can be - e.g. for people that would need to add more error edge cases handlingin the future but not necessarily know about this.

Comment (zrzka) [reply src/builtin/filter/datetime.rs]: @cyplo What kind of people do you mean? I mean, this is an error for the user. How user can add more edge cases handling? I probably don't understand what you do mean.

File: src/error.rs (hunk 2 above)
Comment (zrzka) [reply src/error.rs]: Did add comment about Box. The reason is to keep `Result` as small as possible.

File: src/builtin/filter/datetime.rs (hunk 1 above)
Comment (cyplo) [reply src/builtin/filter/datetime.rs]: yeah, sorry for not being precise here. 
I meant future maintainers of this library, temen. Either me and you in the future, not remembering how to do that, or some other people that are fresh to this codebase.

**Comment by cyplo** [comment]
lgtm :+1:

File: src/builtin/filter/datetime.rs (hunk 1 above)
Comment (zrzka) [reply src/builtin/filter/datetime.rs]: @cyplo Hmm, usage of this simple error should be crystal clear, that's the reason why docs are generated. Not sure if it's worth it, but I did add an example to the `error` module.

screenshot 2018-11-16 at 10 17 08
